main.go

```diff
@@ -42,13 +75,27 @@ func main() {
 
 	assetHandler := http.FileServer(getFileSystem(useFS))
 	e.GET("/*", echo.WrapHandler(assetHandler))
+	e.POST("/api/auth", func(ctx echo.Context) error {
+		credentials := struct {
+			Username string
+			Password string
+		}{}
+		err := ctx.Bind(&credentials)
+		if err != nil {
+			return err
+		}
+		if credentials.Username != config.Credentials.Username|| credentials.Password != config.Credentials.Password {
+			return echo.NewHTTPError(http.StatusForbidden, "invalid credentials")
+		}
+		return ctx.JSON(200, map[string]string{"token": "123"})
```

Comment:
if we keep all state client-side (put it all in the token) we probably don't need a server-side session?

Reply:
When we are planning to open up our demo apps coupled to the development network to the public interwebs we need  real auth. So we do need some form of server sessions.

Reply:
I don't say we don't need auth (we do need, as you correctly point out), just saying that we shouldn't introduce sessions (in near future) when we don't need them